openlibrary/plugins/upstream/utils.py

```diff
@@ -804,8 +805,22 @@ def canonical_url(self):
         readable_path = web.ctx.get('readable_path', web.ctx.path) or ''
         query = web.ctx.query or ''
         host = web.ctx.host or ''
-        url = (host + readable_path + query)
-        return ("https://" + url) if url else ''
+        url = host + readable_path + query
+        if url:
+            url = "https://" + url
+            parsed_url = urlparse(url)
+
+            parsed_query = parse_qs(parsed_url.query)
+            queries_to_exclude = ['sort', 'mode', 'v', 'type', 'debug']
+
+            canonical_query = {q: v for q, v in parsed_query.items() if q not in queries_to_exclude}
+            query = urlencode(canonical_query, doseq=True)
```

Comment:
Careful -- at the top of this file we import `from six.moves.urllib.parse import parse_qs, urlencode` but then we override and redefine `urlencode` (a `@public` method which doesn't take `doseq`) below.

Reply:
I think we may be able to import `urlencode` as `parse_urlencode` and call `parse_urlencode` at L817 to resolve this problem